### DISCUSSION TITLE
Connecting TL021WVC02CT-B1323B via SPI to ESP-WROOM-32

### DISCUSSION THREAD

**Discussion by brightproject** [moononournation/Arduino_GFX#518] [Q&A]
I have a `2.1 inch` display on the `ST7701`controller

![pinout](https://github.com/user-attachments/assets/82f14b8b-6762-4c96-aefd-6305e2c30673)

pinout

![pinout_1](https://github.com/user-attachments/assets/e131dee1-6212-4c3e-bc25-4ed980cc1010)

I also have an `esp-wroom-32`, its pinout is like this

![lolin32-oled-pinout 3](https://github.com/user-attachments/assets/29a39bed-ae26-476f-a4f5-1d72e259770b)

Can I connect this display to this microcontroller via `3-SPI`?
Or only via `4-SPI` or `RGB` interface?

I found the only file in this library that seems to work with the display controller `ST7701`
But here the connection is via the `RGB` interface.

> https://github.com/moononournation/Arduino_GFX/blob/master/src/display/Arduino_RGB_Display.h

The configuration of my equipment that I described above is available to me now.

![hardware](https://github.com/user-attachments/assets/bd04995d-28d2-4aeb-b9da-36a4c49d8f9b)

In general, I have an `esp32-s3-16nr8` and in the future I plan to make a printed circuit board with an `RGB` connection, but now I wanted to try working with this display with an old `esp32` microcontroller.

In a simple example

> https://github.com/moononournation/Arduino_GFX/blob/b3d190810b8e8c6d558b72176a247b2eb811241d/examples/HelloWorld/HelloWorld.ino#L8

I didn't understand how to set my pins for the display, or are they hardcoded in the code and I need to `connect/solder` the contacts as indicated in the commented line?

**Comment by MihaelBercic** [comment]
Did you have any success with getting the display to work? Did you get the backlight to turn on?
